src/objclass/objclass.h

```diff
@@ -4,8 +4,6 @@
 #ifndef CEPH_OBJCLASS_H
 #define CEPH_OBJCLASS_H
 
-#ifdef __cplusplus
```

Comment:
why? assume we need to include this header file in a compiling unit programmed in C language, we should exclude the C++ headers from it, right? presumably that's why we add this guard.

Reply:
You're right. We shouldn't change the scope guard here.